### PR TITLE
Fix undefined initialValue handling in Array.prototype.reduce and reduceRight function

### PR DESCRIPTION
**Pull request by kkristof**
JerryScript-DCO-1.0-Signed-off-by: Kristof Kosztyo kkosztyo.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp [hunk 1]
```diff
@@ -2578,12 +2580,12 @@ ecma_builtin_array_prototype_object_reduce (ecma_value_t this_arg, /**< this arg
...
 
-    JERRY_ASSERT (ecma_is_value_object (arg1));
-    func_object_p = ecma_get_object_from_value (arg1);
+    JERRY_ASSERT (ecma_is_value_object (callbackfn));
+    func_object_p = ecma_get_object_from_value (callbackfn);
     ecma_value_t accumulator = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
 
     /* 5. */
-    if (len_number == ECMA_NUMBER_ZERO && ecma_is_value_undefined (arg2))
+    if (len_number == ECMA_NUMBER_ZERO && ecma_is_value_undefined (initialValue))
```
Comment (egavrin) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp]: Please, avoid usage of CamelCase. `initial_value`

Comment (kkristof) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp]: Sorry, I missed that.

**Comment by kkristof** [comment]
I have updated the PR

**Comment by egavrin** [comment]
Good to me.

**Comment by dbatyai** [comment]
lgtm

**Pull request merged**